packages/patterns-react/src/patterns/TableOfContents/TableOfContents.js

```diff
@@ -0,0 +1,159 @@
+import { TABLE_OF_CONTENTS } from '../../internal/FeatureFlags';
+import { Layout } from '@carbon/ibmdotcom-react';
+import DesktopMenu from '../TableOfContents/DesktopMenu';
+import MobileMenu from '../TableOfContents/MobileMenu';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ * Table of Contents pattern
+ *
+ * @param {object} props props object
+ * @param {object} props.menuItems menu items object
+ * @param {*} props.children children property of component
+ * @returns {*} JSX Object
+ */
+const TableOfContents = ({ menuItems, children, menuLabel }) => {
+  const [selectedId, setSelectedId] = useState(menuItems[0].id);
+  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].title);
+
+  useEffect(() => {
+    scrollStop(setSelectedItem);
+  });
+
+  /**
+   * Check whether provided anchor tags are in visible viewport
+   *
+   * @returns {Array} array of name attributes
+   */
+  const getElemsInView = () => {
+    const eles = document.querySelectorAll('a[name]');
+    let elesInView = [];
+    eles.forEach(element => {
+      const bounding = element.getBoundingClientRect();
+      if (
+        bounding.top >= 0 &&
+        bounding.left >= 0 &&
+        bounding.bottom <=
+          (root.innerHeight || document.documentElement.clientHeight) &&
+        bounding.right <=
+          (root.innerWidth || document.documentElement.clientWidth)
+      ) {
+        elesInView.push(element.getAttribute('name'));
+      }
+    });
+    return elesInView;
+  };
+
+  /**
+   * Set selected id & title
+   *
+   */
+  const setSelectedItem = () => {
+    const elems = getElemsInView();
+    const id = elems[0] || menuItems[0].id;
+    const filteredItems = menuItems.filter(menu => {
+      if (id !== 'undefined') {
+        return menu.id == id;
+      }
+    });
+    const title = filteredItems[0].title;
+    setSelectedId(id);
+    setSelectedTitle(title);
+  };
+
+  /**
+   * Detect scroll stop event and run callback function
+   *
+   * @param {*} callback callback function
+   */
+  const scrollStop = callback => {
+    if (!callback || typeof callback !== 'function') return;
+    let isScrolling;
+    root.addEventListener(
+      'scroll',
+      () => {
+        root.clearTimeout(isScrolling);
+        isScrolling = setTimeout(() => {
+          callback();
+        }, 66);
+      },
+      false
+    );
+  };
+
+  /**
+   *
+   *
+   * @param {*} id selected id of menu item
+   * @param {*} title selected title of menu item
+   */
+  const updateState = (id, title) => {
+    setSelectedId(id);
+    setSelectedTitle(title);
+  };
+
+  const layoutProps = {};
+  layoutProps['type'] = '1-3';
+  layoutProps['marginTop'] = 'none';
+  layoutProps['marginBottom'] = 'none';
+
+  const props = {};
+  props['menuItems'] = menuItems;
+  props['selectedId'] = selectedId;
+  props['selectedTitle'] = selectedTitle;
+  props['menuLabel'] = menuLabel;
+  props['updateState'] = updateState;
```

Comment:
@sannminwin This might be more of a preference thing, but it may be more terse to do it this way:

const props = {
  menuItems,
  selectId,
  selectedTitle,
  menuLabel,
  updateState,
};

Reply:
updated